File: jfingerprintmanager/src/main/java/com/jesusm/jfingerprintmanager/base/ui/SystemImpl.java
package com.jesusm.jfingerprintmanager.base.ui;

import android.support.v4.app.FragmentManager;

public class SystemImpl implements System {
    private static final java.lang.String FINGERPRINT_DIALOG_TAG = "JFingerprintManager:fingerprintDialog";

    private FingerprintBaseDialogFragment fingerprintBaseDialogFragment;
    private FragmentManager fragmentManager;

    @Override
    public void showDialog() {
        fingerprintBaseDialogFragment.show(fragmentManager, FINGERPRINT_DIALOG_TAG);
    }

    @Override
    public void addDialogInfo(FingerprintBaseDialogFragment.Builder builder, FragmentManager fragmentManager) {
        this.fingerprintBaseDialogFragment = builder.build();
        this.fragmentManager = fragmentManager;
    }
}

File: jfingerprintmanager/src/main/kotlin/com/jesusm/jfingerprintmanager/base/ui/SystemImpl.kt
package com.jesusm.jfingerprintmanager.base.ui

import android.support.v4.app.FragmentManager

class SystemImpl : System {
    private val FINGERPRINT_DIALOG_TAG = "JFingerprintManager:fingerprintDialog"

    private var fingerprintBaseDialogFragment: FingerprintBaseDialogFragment<*>? = null
    private var dialogFragmentManager: FragmentManager? = null

    override fun showDialog() {
        fingerprintBaseDialogFragment?.show(dialogFragmentManager, FINGERPRINT_DIALOG_TAG)
    }

    override fun addDialogInfo(builder: FingerprintBaseDialogFragment.Builder<out FingerprintBaseDialogFragment<*>, *>?,
                               fragmentManager: FragmentManager?) {
        builder?.let {
            fingerprintBaseDialogFragment = it.build()
            dialogFragmentManager = fragmentManager
        }
    }
}